### PR TITLE
refactor: refactor carousel to support namespace

### PR DESCRIPTION
**Pull request by iamkun**
Please make sure these boxes are checked before submitting your PR, thank you!
ref #5374
- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/carousel/src/item.vue [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-      'is-animating': data.animating,
-    }"
+    :class="[
+      ns.e('item'),
+      {
+        'is-active': data.active,
+        'is-in-stage': data.inStage,
+        'is-hover': data.hover,
+        'is-animating': data.animating,
+      },
```
Comment (sxzz) [inline packages/components/carousel/src/item.vue:11]: ns.is('active', data.active)

File: packages/components/carousel/src/item.vue [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+      type === 'card' ? ns.em('item', 'card') : '',
```
Comment (sxzz) [inline packages/components/carousel/src/item.vue:12]: {[ns.em('item', 'card')]: type === 'card'}

**Comment by iamkun** [comment]
fixed

File: packages/components/carousel/src/item.vue [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+      { [ns.em('item', 'card')]: type === 'card' },
+    ]"
     :style="itemStyle"
     @click="handleItemClick"
   >
     <div
       v-if="type === 'card'"
       v-show="!data.active"
-      class="el-carousel__mask"
+      :class="ns.b('mask')"
```
Comment (sxzz) [inline packages/components/carousel/src/item.vue:18]:
```suggestion
      :class="ns.e('mask')"
```

**Review by sxzz — CHANGES_REQUESTED** [review]

File: packages/components/carousel/src/item.vue (hunk 3 above)
Comment (iamkun) [reply packages/components/carousel/src/item.vue]: my bad, fixed

**Review by sxzz — APPROVED** [review]

**Review by sxzz — APPROVED** [review]

**Pull request merged**